crates/bevy_render/src/camera/visible_entities.rs

```diff
@@ -24,22 +24,92 @@ impl VisibleEntities {
     }
 }
 
+/// A mask that describes which rendering group an entity belongs to.
+/// Cameras with this component will only render entities with a matching
+/// mask.
+#[derive(Copy, Clone, Debug, Reflect, PartialEq, Eq, PartialOrd, Ord)]
+#[reflect(Component)]
+pub struct RenderingMask(pub u32);
+
+impl Default for RenderingMask {
+    fn default() -> Self {
+        RenderingMask(0)
```

Comment:
I think having a default mask at 1 would be better
A default at 0 means the default will match only 0, but 1 would match everything with bit 1 on, which seems better for the default camera? Not entirely sure, what do you think?

Reply:
**tl;dr** Either of these change requests make sense on their own but the two together alter the logic I intended.

What the default is seems a bit arbitrary to me but I do think that "no mask" (`RenderingMask(0)`) is what entities _without_ this component should be equal to. 
The suggestion that the default be mask `1` (or group `0`) has implications about your comment above. We only want cameras with masks rendering entities with matching masks. If the camera has a mask but the entity does not, we continue the loop. If the entity has a mask but the camera does not, we continue the loop. If they both have masks but they don't match, we continue the loop.
Consider the case where we have a camera with mask `1` (group `0`) and an entity with no mask. If the default is mask `1` then the gating logic would say "these both match, since the camera defaults to mask `1`, group `0`". But what we really want is for the loop to continue since the camera has no mask at all.

If we change the default to mask `1`, group `0`, then I think the gating logic should still use `unwrap_or_else(|| RenderingMask(0))`. 

What do you think @mockersf ?

Reply:
My opinions were more:

- by default, the camera should display everything, so it makes sense to be `group(0)`
- by default, all entities should be displayed, so they should be `group(0)`
- not wanting to display an entity should be an extra step, so setting its mask to `RenderingMask::default().without_group(0)`. Maybe there could be a shorter method for that

I think that makes more sense if all cameras have a mask, but if they have no mask unless the user add one yours is better

Reply:
Ah, yes - I think that's where the confusion comes from. Being in `group(0)` means that an entity will only be rendered by cameras in `group(0)`. A camera in `group(0)` will only render entities in `group(0)`. **Without** a mask component a camera will render all entities without a mask. In a way not having a mask is a bit like being in the default group. For this reason making the default mask "empty" (`RenderingMask(0)`) makes the comparison logic very simple. 

I'm expecting folks will only use `RenderingMask` if they have a special camera that only renders some objects. For example - if you want to build a 3d user interface with a different projection than the rest of your scene. For this reason most cameras will _not_ have a mask.